teps/0063-workspace-dependencies.md

```diff
@@ -0,0 +1,233 @@
+[TEP-0050 Ignore Task Failures](https://github.com/tektoncd/community/blob/main/teps/0050-ignore-task-failures.md) and 
+[TEP-0059 Skip Guarded Task Only](https://github.com/tektoncd/community/blob/main/teps/0059-skip-guarded-task-only.md). 
+These TEPs will unblock the execution of dependent `Tasks` when a parent `Task` fails or is skipped. In those scenarios, 
+ordering-dependent `Tasks` should execute successfully while resource-dependent `Tasks` should be skipped because of
+missing resources. However, because users cannot specify resource dependencies based on `Workspaces`, resource-dependent 
+`Tasks` based on `Workspaces` will be executed and they won't be able to resolve missing resources that they expect in 
+the `Workspaces`, so they will fail. As a result, the whole `Pipeline` would fail making the failure and skipping 
+strategies not work as expected. Thus, we need to enable users to specify resource dependencies based on `Workspaces`.  
+
+### Goals
+
+<!--
+List the specific goals of the TEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+The main goal of this TEP is to enable users to specify resource dependencies based on `Workspaces`. 
+
+### Non-Goals
+
+<!--
+What is out of scope for this TEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+Specifying the files that a `Task` produces to or expects from a `Workspace` is out of scope for this TEP and is 
+addressed in [TEP-0030 Workspace Paths](https://github.com/tektoncd/community/blob/main/teps/0030-workspace-paths.md).
+
+### Use Cases
+
+<!--
+Describe the concrete improvement specific groups of users will see if the
+Motivations in this doc result in a fix or feature.
+
+Consider both the user's role (are they a Task author? Catalog Task user?
+Cluster Admin? etc...) and experience (what workflows or actions are enhanced
+if this problem is solved?).
+-->
+
+Let's take the manual approval use case described in further detail in [TEP-0059 Skip Guarded Task Only](https://github.com/tektoncd/community/blob/main/teps/0059-skip-guarded-task-only.md).
+
+A user needs to design a `Pipeline` with a _manual approval_ `Task` that is executed when merging a pull request only. 
+The execution of the _manual approval_ `Task` is guarded using `WhenExpressions`. To reuse the same `Pipeline` when 
+merging and not merging, the user needs the dependent `Tasks` to execute when the guarded _manual approval_ `Task` is 
+skipped.
+
+```
+          lint                     unit-tests
+           |                           |
+           v                           v
+   report-linter-output        integration-tests
+                                       |
+                                       v
+                                 manual-approval
+                                 |            |
+                                 v        (approver)
```

Comment:
i think (approver) represents a result? i thought it was a task or when expression looking at this diagram and had to read below. maybe it would be a bit clearer to flesh out these tasks a bit more, e.g. something like:

```
task: manual-approval
when: merging
result: approver
```

```
task: slack-msg
needs approver result / or explain params / something
```

Reply:
(it IS explained clearly below tho, so feel free to ignore this if you feel like it makes the diagram less clear or isn't worth the work)

Reply:
wanted to keep the diagram simple, so added further detail about `manual-approval` and `slack-mgs` right before describing the failure/skipping strategies, thanks :)